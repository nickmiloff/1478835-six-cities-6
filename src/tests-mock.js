export const simpleMock = `test`;

export const cards = {
  changedCard: {
    id: 2,
    isFavorite: true
  },
  start: [
    {
      id: 1,
      isFavorite: true
    },
    {
      id: 2,
      isFavorite: false
    },
    {
      id: 3,
      isFavorite: false
    }
  ],
  end: [
    {
      id: 1,
      isFavorite: true
    },
    {
      id: 2,
      isFavorite: true
    },
    {
      id: 3,
      isFavorite: false
    }
  ],
  endWithRemove: [
    {
      id: 1,
      isFavorite: true
    },
    {
      id: 3,
      isFavorite: false
    }
  ]
};

export const auth = {
  api: {
    "id": 1,
    "email": `mail@mail.com`,
    "name": `name`,
    "avatar_url": `avatar.jpeg`,
    "is_pro": true
  },
  adapt: {
    id: 1,
    email: `mail@mail.com`,
    name: `name`,
    avatarUrl: `avatar.jpeg`,
    isPro: true
  }
};

export const user = {
  email: `mail@mail.com`,
  password: 12345678
};

export const cardsFull = {
  api: [
    {
      "city": {
        "name": `Paris`,
        "location": {
          "latitude": 48.85661,
          "longitude": 2.351499,
          "zoom": 13
        }
      },
      "images": [],
      "title": `Tile House`,
      "rating": 2,
      "type": `house`,
      "bedrooms": 2,
      "price": 161,
      "goods": [`Laptop friendly workspace`],
      "host": {
        "id": 25,
        "name": `Angelina`,
        "avatar_url": `img/avatar-angelina.jpg`,
        "is_pro": true
      },
      "description": `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
      "location": {
        "latitude": 48.85761,
        "longitude": 2.358499,
        "zoom": 16
      },
      "id": 1,
      "is_favorite": false,
      "is_premium": true,
      "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
      "max_adults": 9
    },
    {
      "city": {
        "name": `Amsterdam`,
        "location": {
          "latitude": 52.37454,
          "longitude": 4.897976,
          "zoom": 13
        }
      },
      "images": [],
      "title": `Loft Studio in the Central Area`,
      "rating": 3.6,
      "type": `apartment`,
      "bedrooms": 3,
      "price": 271,
      "goods": [
        `Laptop friendly workspace`,
        `Washer`,
        `Breakfast`,
        `Air conditioning`
      ],
      "host": {
        "id": 25,
        "name": `Angelina`,
        "avatar_url": `img/avatar-angelina.jpg`,
        "is_pro": true
      },
      "description": `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      "location": {
        "latitude": 52.388540000000006,
        "longitude": 4.899976,
        "zoom": 16
      },
      "id": 2,
      "is_favorite": false,
      "is_premium": false,
      "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
      "max_adults": 8
    },
    {
      "city": {
        "name": `Paris`,
        "location": {
          "latitude": 48.85661,
          "longitude": 2.351499,
          "zoom": 13
        }
      },
      "images": [],
      "title": `Tile House`,
      "rating": 4.6,
      "type": `house`,
      "bedrooms": 2,
      "price": 865,
      "goods": [`Laptop friendly workspace`, `Breakfast`],
      "host": {
        "id": 25,
        "name": `Angelina`,
        "avatar_url": `img/avatar-angelina.jpg`,
        "is_pro": true
      },
      "description": `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      "location": {
        "latitude": 48.83861,
        "longitude": 2.350499,
        "zoom": 16
      },
      "id": 3,
      "is_favorite": true,
      "is_premium": false,
      "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
      "max_adults": 8
    }
  ],
  adapted: [
    {
      city: {
        name: `Paris`,
        location: {
          latitude: 48.85661,
          longitude: 2.351499,
          zoom: 13,
        },
      },
      images: [],
      title: `Tile House`,
      rating: 2,
      type: `house`,
      bedrooms: 2,
      price: 161,
      goods: [`Laptop friendly workspace`],
      host: {
        id: 25,
        name: `Angelina`,
        avatarUrl: `img/avatar-angelina.jpg`,
        isPro: true,
      },
      description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
      location: {
        latitude: 48.85761,
        longitude: 2.358499,
        zoom: 16,
      },
      id: 1,
      isFavorite: false,
      isPremium: true,
      previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
      maxAdults: 9,
    },
    {
      city: {
        name: `Amsterdam`,
        location: {
          latitude: 52.37454,
          longitude: 4.897976,
          zoom: 13,
        },
      },
      images: [],
      title: `Loft Studio in the Central Area`,
      rating: 3.6,
      type: `apartment`,
      bedrooms: 3,
      price: 271,
      goods: [
        `Laptop friendly workspace`,
        `Washer`,
        `Breakfast`,
        `Air conditioning`,
      ],
      host: {
        id: 25,
        name: `Angelina`,
        avatarUrl: `img/avatar-angelina.jpg`,
        isPro: true,
      },
      description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      location: {
        latitude: 52.388540000000006,
        longitude: 4.899976,
        zoom: 16,
      },
      id: 2,
      isFavorite: false,
      isPremium: false,
      previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
      maxAdults: 8,
    },
    {
      city: {
        name: `Paris`,
        location: {
          latitude: 48.85661,
          longitude: 2.351499,
          zoom: 13,
        },
      },
      images: [],
      title: `Tile House`,
      rating: 4.6,
      type: `house`,
      bedrooms: 2,
      price: 865,
      goods: [`Laptop friendly workspace`, `Breakfast`],
      host: {
        id: 25,
        name: `Angelina`,
        avatarUrl: `img/avatar-angelina.jpg`,
        isPro: true,
      },
      description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      location: {
        latitude: 48.83861,
        longitude: 2.350499,
        zoom: 16,
      },
      id: 3,
      isFavorite: true,
      isPremium: false,
      previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
      maxAdults: 8,
    },
  ]
};

export const comments = {
  api: [
    {
      "comment": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
      "date": `2019-05-08T14:13:56.569Z`,
      "id": 1,
      "rating": 4,
      "user": {
        "avatar_url": `img/1.png`,
        "id": 4,
        "is_pro": false,
        "name": `Max`
      }
    },
    {
      "comment": `A quiet of Amsterdam.`,
      "date": `2019-06-08T14:13:56.569Z`,
      "id": 2,
      "rating": 3,
      "user": {
        "avatar_url": `img/2.png`,
        "id": 5,
        "is_pro": false,
        "name": `Vadim`
      }
    }
  ],
  adapted: [
    {
      comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
      date: `2019-05-08T14:13:56.569Z`,
      id: 1,
      rating: 4,
      user: {
        avatarUrl: `img/1.png`,
        id: 4,
        isPro: false,
        name: `Max`
      }
    },
    {
      comment: `A quiet of Amsterdam.`,
      date: `2019-06-08T14:13:56.569Z`,
      id: 2,
      rating: 3,
      user: {
        avatarUrl: `img/2.png`,
        id: 5,
        isPro: false,
        name: `Vadim`
      }
    }
  ]
};
