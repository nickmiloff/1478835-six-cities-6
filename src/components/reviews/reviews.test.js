import React from 'react';
import {render} from '@testing-library/react';
import Test from '../test/test';

import Reviews from './reviews';
import {comments} from '../../tests-mock';

describe(`Reviews component test`, () => {
  it(`Reviews component' should render correctly with authorized user`, () => {
    const {container} = render(
        <Test initial={false}>
          <Reviews reviews={comments.adapted} isAuth={true} />
        </Test>
    );

    expect(container).toMatchSnapshot();
  });

  it(`Reviews component' should render correctly with unauthorized user`, () => {
    const {container} = render(
        <Test>
          <Reviews reviews={comments.adapted} isAuth={false} />
        </Test>
    );

    expect(container).toMatchSnapshot();
  });
});
