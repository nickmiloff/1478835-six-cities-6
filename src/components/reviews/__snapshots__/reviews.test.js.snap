// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reviews component test Reviews component' should render correctly with authorized user 1`] = `
<div>
  <section
    class="property__reviews reviews"
  >
    <h2
      class="reviews__title"
    >
      Reviews · 
      <span
        class="reviews__amount"
      >
        2
      </span>
    </h2>
    <ul
      class="reviews__list"
    >
      <li
        class="reviews__item"
      >
        <div
          class="reviews__user user"
        >
          <div
            class="reviews__avatar-wrapper user__avatar-wrapper"
          >
            <img
              alt="Reviews avatar"
              class="reviews__avatar user__avatar"
              height="54"
              src="img/1.png"
              width="54"
            />
          </div>
          <span
            class="reviews__user-name"
          >
            Max
          </span>
        </div>
        <div
          class="reviews__info"
        >
          <div
            class="reviews__rating rating"
          >
            <div
              class="reviews__stars rating__stars"
            >
              <span
                style="width: 80%;"
              />
              <span
                class="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <p
            class="reviews__text"
          >
            A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
          </p>
          <time
            class="reviews__time"
            datetime="2019-05-08T14:13:56.569Z"
          >
            May 2019
          </time>
        </div>
      </li>
      <li
        class="reviews__item"
      >
        <div
          class="reviews__user user"
        >
          <div
            class="reviews__avatar-wrapper user__avatar-wrapper"
          >
            <img
              alt="Reviews avatar"
              class="reviews__avatar user__avatar"
              height="54"
              src="img/2.png"
              width="54"
            />
          </div>
          <span
            class="reviews__user-name"
          >
            Vadim
          </span>
        </div>
        <div
          class="reviews__info"
        >
          <div
            class="reviews__rating rating"
          >
            <div
              class="reviews__stars rating__stars"
            >
              <span
                style="width: 60%;"
              />
              <span
                class="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <p
            class="reviews__text"
          >
            A quiet of Amsterdam.
          </p>
          <time
            class="reviews__time"
            datetime="2019-06-08T14:13:56.569Z"
          >
            June 2019
          </time>
        </div>
      </li>
    </ul>
    <form
      action="#"
      class="reviews__form form"
      data-testid="form"
      method="post"
    >
      <label
        class="reviews__label form__label"
        for="review"
      >
        Your review
      </label>
      <div
        class="reviews__rating-form form__rating"
      >
        <input
          class="form__rating-input visually-hidden"
          id="5-stars"
          name="rating"
          type="radio"
          value="5"
        />
        <label
          class="reviews__rating-label form__rating-label"
          for="5-stars"
          title="perfect"
        >
          <svg
            class="form__star-image"
            height="33"
            width="37"
          >
            <use
              xlink:href="#icon-star"
            />
          </svg>
        </label>
        <input
          class="form__rating-input visually-hidden"
          id="4-stars"
          name="rating"
          type="radio"
          value="4"
        />
        <label
          class="reviews__rating-label form__rating-label"
          for="4-stars"
          title="perfect"
        >
          <svg
            class="form__star-image"
            height="33"
            width="37"
          >
            <use
              xlink:href="#icon-star"
            />
          </svg>
        </label>
        <input
          class="form__rating-input visually-hidden"
          id="3-stars"
          name="rating"
          type="radio"
          value="3"
        />
        <label
          class="reviews__rating-label form__rating-label"
          for="3-stars"
          title="perfect"
        >
          <svg
            class="form__star-image"
            height="33"
            width="37"
          >
            <use
              xlink:href="#icon-star"
            />
          </svg>
        </label>
        <input
          class="form__rating-input visually-hidden"
          id="2-stars"
          name="rating"
          type="radio"
          value="2"
        />
        <label
          class="reviews__rating-label form__rating-label"
          for="2-stars"
          title="perfect"
        >
          <svg
            class="form__star-image"
            height="33"
            width="37"
          >
            <use
              xlink:href="#icon-star"
            />
          </svg>
        </label>
        <input
          class="form__rating-input visually-hidden"
          id="1-stars"
          name="rating"
          type="radio"
          value="1"
        />
        <label
          class="reviews__rating-label form__rating-label"
          for="1-stars"
          title="perfect"
        >
          <svg
            class="form__star-image"
            height="33"
            width="37"
          >
            <use
              xlink:href="#icon-star"
            />
          </svg>
        </label>
      </div>
      <textarea
        class="reviews__textarea form__textarea"
        data-testid="review"
        id="review"
        maxlength="300"
        name="review"
        placeholder="Tell how was your stay, what you like and what can be improved"
      />
      <div
        class="reviews__button-wrapper"
      >
        <p
          class="reviews__help"
        >
          To submit review please make sure to set 
          <span
            class="reviews__star"
          >
            rating
          </span>
           and describe your stay with at least 
          <b
            class="reviews__text-amount"
          >
            50 characters
          </b>
          .
        </p>
        <button
          class="reviews__submit form__submit button"
          disabled=""
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </section>
</div>
`;

exports[`Reviews component test Reviews component' should render correctly with unauthorized user 1`] = `
<div>
  <section
    class="property__reviews reviews"
  >
    <h2
      class="reviews__title"
    >
      Reviews · 
      <span
        class="reviews__amount"
      >
        2
      </span>
    </h2>
    <ul
      class="reviews__list"
    >
      <li
        class="reviews__item"
      >
        <div
          class="reviews__user user"
        >
          <div
            class="reviews__avatar-wrapper user__avatar-wrapper"
          >
            <img
              alt="Reviews avatar"
              class="reviews__avatar user__avatar"
              height="54"
              src="img/1.png"
              width="54"
            />
          </div>
          <span
            class="reviews__user-name"
          >
            Max
          </span>
        </div>
        <div
          class="reviews__info"
        >
          <div
            class="reviews__rating rating"
          >
            <div
              class="reviews__stars rating__stars"
            >
              <span
                style="width: 80%;"
              />
              <span
                class="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <p
            class="reviews__text"
          >
            A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
          </p>
          <time
            class="reviews__time"
            datetime="2019-05-08T14:13:56.569Z"
          >
            May 2019
          </time>
        </div>
      </li>
      <li
        class="reviews__item"
      >
        <div
          class="reviews__user user"
        >
          <div
            class="reviews__avatar-wrapper user__avatar-wrapper"
          >
            <img
              alt="Reviews avatar"
              class="reviews__avatar user__avatar"
              height="54"
              src="img/2.png"
              width="54"
            />
          </div>
          <span
            class="reviews__user-name"
          >
            Vadim
          </span>
        </div>
        <div
          class="reviews__info"
        >
          <div
            class="reviews__rating rating"
          >
            <div
              class="reviews__stars rating__stars"
            >
              <span
                style="width: 60%;"
              />
              <span
                class="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <p
            class="reviews__text"
          >
            A quiet of Amsterdam.
          </p>
          <time
            class="reviews__time"
            datetime="2019-06-08T14:13:56.569Z"
          >
            June 2019
          </time>
        </div>
      </li>
    </ul>
    
  </section>
</div>
`;
